### PR TITLE
Add namespacing

### LINKED ISSUE #35: Add namespacing
Users should be able to specify a namespace for their tags, like `<o:some-tag />`, and hopefully even be able to mix tags from different namespaces. For example, `<l:bind>` for larceny tags, `<wp:posts>` for Offset tags, `<df:form>` for digestive functors, `<myapp:username>` for my app.

### PR DESCRIPTION
**Pull request by emhoracek**
To fix #35

### CODE REVIEW

File: tools/Convert0.2-0.3.hs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    -- these are the blank and plain elements
+    -- if it's in the "svg" namespace, it's a plain node
+    -- otherwise, if there's a namespace, it's definitely a Blank
+    -- if there's not a namespace, and the tag is a member of the set of plain nodes, it's not a Blank
+    -- otherwise, it's a blan
+    Name (Just "svg") name ->
+      NodeElement (PlainElement (Name (Just "svg") name) attrs larcenyNodes)
+    Name (Just ns) name ->
+      NodeElement (BlankElement (Name (Just ns) name) attrs larcenyNodes)
+    Name ns name | HS.member name (svgNodes `HS.difference` html5Nodes) ->
```
Comment (emhoracek) [inline tools/Convert0.2-0.3.hs]: If the tag is in the svgNodes but not html5Nodes, then prefix with "l:". That's the only difference between this script's `toLarcenyNode'` and the internal `toLarcenyNode`.

File: src/Web/Larceny/Internal.hs [hunk 2]
```diff
@@ -33,16 +31,64 @@ parse = parseWithOverrides defaultOverrides
...
+    -- these are the blank and plain elements
+    -- if it's in the "svg" namespace, it's a plain node
+    -- otherwise, if there's a namespace, it's definitely a Blank
+    -- if there's not a namespace, and the tag is a member of the set of plain nodes, it's not a Blank
+    -- otherwise, it's a blan
+    Name (Just "svg") name ->
+      NodeElement (PlainElement (Name (Just "svg") name) attrs larcenyNodes)
+    Name (Just ns) name ->
+      NodeElement (BlankElement (Name (Just ns) name) attrs larcenyNodes)
+    Name ns name | HS.member name allPlainNodes ->
```
Comment (dhartunian) [inline src/Web/Larceny/Internal.hs]: ns should be `Nothing`

File: src/Web/Larceny/Internal.hs [hunk 3]
```diff
@@ -33,16 +31,64 @@ parse = parseWithOverrides defaultOverrides
...
+      NodeElement (BlankElement (Name Nothing "apply-content") attrs larcenyNodes)
+
+    -- these are the blank and plain elements
+    -- if it's in the "svg" namespace, it's a plain node
+    -- otherwise, if there's a namespace, it's definitely a Blank
+    -- if there's not a namespace, and the tag is a member of the set of plain nodes, it's not a Blank
+    -- otherwise, it's a blan
+    Name (Just "svg") name ->
+      NodeElement (PlainElement (Name (Just "svg") name) attrs larcenyNodes)
+    Name (Just ns) name ->
```
Comment (dhartunian) [inline src/Web/Larceny/Internal.hs]: change ns to `"l"`

File: src/Web/Larceny/Internal.hs [hunk 4]
```diff
@@ -33,16 +31,64 @@ parse = parseWithOverrides defaultOverrides
...
+      allPlainNodes = (HS.fromList (customPlainNodes o) `HS.union` html5Nodes `HS.union` svgNodes)
+                             `HS.difference` HS.fromList (overrideNodes o)in
+  case toLarcenyName tn of
+
+    -- these are our special larceny elements
+    Name Nothing "bind" ->
+      NodeElement (BindElement attrs larcenyNodes)
+    Name Nothing "apply" ->
+      NodeElement (ApplyElement attrs larcenyNodes)
+    Name Nothing "apply-content" ->
```
Comment (dhartunian) [inline src/Web/Larceny/Internal.hs]: consider removing special case...??

Comment (dhartunian) [inline src/Web/Larceny/Internal.hs]: nvm

File: src/Web/Larceny/Internal.hs [hunk 5]
```diff
@@ -99,56 +143,62 @@ add :: Substitutions s -> Template s -> Template s
...
-      X.NodeComment c                            -> return ["<!--" <> c <> "-->"]
-      X.NodeInstruction _                        -> return []
+      NodeElement (ApplyElement atr kids) ->
+          processApply atr kids
+      NodeElement (PlainElement tn atr kids) ->
+          processPlain tn atr kids
+      NodeElement (BlankElement (Name Nothing name) atr kids) ->
+          processBlank name atr kids
+      NodeElement (BlankElement (Name (Just _) name) atr kids) ->
+          processBlank name atr kids
```
Comment (dhartunian) [inline src/Web/Larceny/Internal.hs]: this and above case can be combined?

**Pull request merged**